tests/chainer_tests/dataset_tests/test_convert.py

```diff
@@ -0,0 +1,197 @@
+import unittest
+
+import numpy
+
+from chainer import cuda
+from chainer import dataset
+from chainer import testing
+from chainer.testing import attr
+
+
+class TestConcatExamples(unittest.TestCase):
+
+    def get_arrays_to_concat(self, xp):
+        return [xp.random.rand(2, 3) for _ in range(5)]
+
+    def check_concat_arrays(self, arrays, device=None):
+        array = dataset.concat_examples(arrays, device)
+        self.assertEqual(array.shape, (len(arrays),) + arrays[0].shape)
```

Comment:
We should check result of the `concat_examples` (namely, `array`) is correctly sent to the device specified by `device`.